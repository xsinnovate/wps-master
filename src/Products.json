{
  "prod1": {
    "name": "Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red",
    "image": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
    "price": 102.96,
    "pickup": 3.85
  }
}
