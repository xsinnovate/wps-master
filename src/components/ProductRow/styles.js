import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  photo: {
    flex: 1,
  },
  image: {
    flex: 1,
    resizeMode: 'contain',
  },
  product: {
    flex: 2,
  },
  details: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: 6,
  },
  price: {
    paddingBottom: 4,
  },
  priceText: {
    fontWeight: '600',
  },
  priceOriginal: {
    color: '#999',
    fontWeight: '600',
    textDecorationLine: 'line-through',
  },
  quantity: {
    fontSize: 13,
  },
});
