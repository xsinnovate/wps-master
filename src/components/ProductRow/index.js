import PropTypes from 'prop-types';
import React from 'react';
import { Image, Text, View } from 'react-native';
import styles from './styles';

export default function ProductRow({ product }) {
  const {
    image,
    name,
    price,
    pickup,
    quantity,
  } = product;

  return (
    <View style={styles.container}>
      <View style={styles.photo}>
        <Image
          source={{ uri: image }}
          style={styles.image}
        />
      </View>
      <View style={styles.product}>
        <Text>
          {name}
        </Text>
        <View style={styles.details}>
          <View>
            <View style={styles.price}>
              <Text style={styles.priceText}>
                {`$${(price - pickup)}`}
              </Text>
            </View>
            <View>
              <Text style={styles.priceOriginal}>
                {`$${price}`}
              </Text>
            </View>
          </View>
          <View>
            <Text style={styles.quantity}>
              {`Qty: ${quantity}`}
            </Text>
          </View>
        </View>
      </View>
    </View>
  );
}

ProductRow.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    pickup: PropTypes.number.isRequired,
    quantity: PropTypes.number.isRequired,
  }).isRequired,
};
