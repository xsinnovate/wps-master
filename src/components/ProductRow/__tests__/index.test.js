import React from 'react';
import renderer from 'react-test-renderer';
import Products from '../../../Products.json';
import ProductRow from '..';

describe('<ProductRow />', () => {
  it('renders correctly', () => {
    const tree = renderer.create((
      <ProductRow
        product={{
          ...Products.prod1,
          quantity: 1,
        }}
      />
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
