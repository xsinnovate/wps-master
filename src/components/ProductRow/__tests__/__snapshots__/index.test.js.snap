// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRow /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i5.walmartimages.com/asr/e73e1252-642c-4473-93ea-fd3b564a7027_1.3e81ea58fa3042452fe185129a4a865f.jpeg?odnHeight=450&odnWidth=450&odnBg=FFFFFF",
        }
      }
      style={
        Object {
          "flex": 1,
          "resizeMode": "contain",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <Text>
      Essentials by OFM ESS-3085 Racing Style Leather Gaming Chair, Red
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 6,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "paddingBottom": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "fontWeight": "600",
              }
            }
          >
            $99.11
          </Text>
        </View>
        <View>
          <Text
            style={
              Object {
                "color": "#999",
                "fontWeight": "600",
                "textDecorationLine": "line-through",
              }
            }
          >
            $102.96
          </Text>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 13,
            }
          }
        >
          Qty: 1
        </Text>
      </View>
    </View>
  </View>
</View>
`;
